internal/controller/storageclient_controller.go

```diff
@@ -438,6 +439,9 @@ func (r *storageClientReconcile) deleteStorageClaims(claimName string) error {
 	storageClaim.Name = claimName
 
 	if err := r.Client.Get(r.ctx, types.NamespacedName{Name: claimName}, storageClaim); client.IgnoreNotFound(err) != nil {
+		if meta.IsNoMatchError(err) {
+			return nil
+		}
 		return fmt.Errorf("Failed to get StorageClaim %q: %w", claimName, err)
```

Comment:
this is introducing matching of specific errors in two places, can be written as below still maintaining single positive exit at the end
```suggestion
	if err := r.Client.Get(r.ctx, types.NamespacedName{Name: claimName}, storageClaim); if err != nil {
		if !(meta.IsNoMatchError(err) || kerrors.IsNotFound(err)) {
                    return fmt.Errorf("Failed to get StorageClaim %q: %w", claimName, err)
		}	
```